internal/workflow/preprocessing.go

```diff
@@ -112,6 +118,64 @@ func (w *PreprocessingWorkflow) Execute(
 
 	localPath := filepath.Join(w.sharedPath, filepath.Clean(params.RelativePath))
 
+	if w.checkDuplicates {
+		// Calculate SIP checksum.
+		ev := result.newEvent(ctx, "Calculate SIP checksum")
```

Comment:
Help with event names and outcomes would be appreciated, I got lost between "duplicate/duplicates/duplicated SIP" and went with basic event names and the same name plus passed/failed in the outcome.